include/jwt-cpp/jwt.h

```diff
@@ -2606,12 +2607,166 @@ namespace jwt {
+				bool matches = false;
+				switch (expected.get_type()) {
+				case json::type::boolean: matches = expected.as_bool() == jc.as_bool(); break;
+				case json::type::integer: matches = expected.as_int() == jc.as_int(); break;
+				case json::type::number: matches = expected.as_number() == jc.as_number(); break;
+				case json::type::string: matches = expected.as_string() == jc.as_string(); break;
+				case json::type::array:
+				case json::type::object:
+					matches = json_traits::serialize(expected.to_json()) == json_traits::serialize(jc.to_json());
+					break;
+				default: throw std::logic_error("internal error, should be unreachable");
+				}
+				if (!matches) {
+					ec = error::token_verification_error::claim_value_missmatch;
+					return;
+				}
+			}
+		};
+
+		/**
+		 * Checks that the current time is before the time specified in the given
+		 * claim. This is identical to how the "exp" check works.
+		 */
+		template<typename json_traits>
+		struct date_before_claim {
+			size_t leeway;
+			void operator()(const verify_context<json_traits>& ctx, std::error_code& ec) {
+				auto c = ctx.jwt.get_payload_claim(ctx.claim_key).as_date();
+				if (ctx.current_time > c + std::chrono::seconds(leeway)) {
+					ec = error::token_verification_error::token_expired;
+				}
+			}
+		};
+
+		/**
+		 * Checks that the current time is after the time specified in the given
+		 * claim. This is identical to how the "nbf" and "iat" check works.
+		 */
+		template<typename json_traits>
+		struct date_after_claim {
+			size_t leeway;
+			void operator()(const verify_context<json_traits>& ctx, std::error_code& ec) {
+				auto c = ctx.jwt.get_payload_claim(ctx.claim_key).as_date();
+				if (ctx.current_time < c - std::chrono::seconds(leeway)) {
+					ec = error::token_verification_error::token_expired;
+				}
+			}
+		};
+
+		/**
+		 * Checks if the given set is a subset of the set inside the token.
+		 * If the token value is a string it is traited as a set of a single element.
+		 * The comparison is case sensitive.
+		 */
+		template<typename json_traits>
+		struct is_subset_claim {
+			typename basic_claim<json_traits>::set_t expected;
+			void operator()(const verify_context<json_traits>& ctx, std::error_code& ec) {
+				if (!ctx.jwt.has_payload_claim(ctx.claim_key)) {
+					ec = error::token_verification_error::audience_missmatch;
+					return;
+				}
+				auto c = ctx.jwt.get_payload_claim(ctx.claim_key);
+				if (c.get_type() == json::type::string) {
+					if (expected.size() > 1 || !(expected.empty() || *expected.begin() == c.as_string())) {
+						ec = error::token_verification_error::audience_missmatch;
+						return;
+					}
+				} else {
+					auto jc = c.as_set();
+					for (auto& e : expected) {
+						if (jc.find(e) == jc.end()) {
+							ec = error::token_verification_error::audience_missmatch;
+							return;
+						}
+					}
+				}
+			}
+		};
+
+		/**
+		 * Checks if the claim is a string and does an case insensitive comparison.
```

Comment:
Doing this correctly would involve utf-8 handling or am I wrong ?
codecvt should be able to do this in an platform independed way, however its deprecated since C++17 (but will remain in the standard until a replacement is found).
However even with codecvt it means having to deal with locales and weird edge cases, which doesn't sound fun to me.

Reply:
💯 I agree, I think just adding a small disclaimer is more then enough